geoapps/inversion/potential_fields/application.py

```diff
@@ -1252,12 +1272,27 @@ def write_trigger(self, _):
                             value = value.value
                         if isinstance(value, uuid.UUID):
                             value = new_workspace.get_entity(value)[0]
-                        setattr(self.params, sub_key, value)
 
-            self.params.write_input_file(
+                        if hasattr(self.params, sub_key):
+                            param_dict[sub_key.lstrip("_")] = value
+
+            # Create new params object and write
+            ifile = InputFile(
+                ui_json=self.params.input_file.ui_json,
+                validation_options={"disabled": True},
+                workspace=new_workspace,
+            )
+            ifile.data.update(param_dict)
+
+            new_params = self.params.__class__(input_file=ifile)
+            new_params.write_input_file(
                 name=self._ga_group_name.value + ".ui.json",
                 path=self.export_directory.selected_path,
             )
+            ifile.name = self._ga_group_name.value + ".ui.json"
+            ifile.path = self.export_directory.selected_path
+            self.params.input_file = ifile
```

Comment:
Just a note that I will make changes to avoid this. We don't want to replace the self.params while the application is running as it would be pointing to a new workspace with less entities. This would have consequences on the widgets.